File: OtherLanguages/PHP/CrashReporter/Data/DateAndTime.php
<?
class DateAndTime
{
	public $year;
	public $month;
	public $day;
	public $hour;
	public $minute;
	public $second;

	public function read($reader)
	{
		$this->year = $reader->readInteger(0);
		$this->month = $reader->readInteger(1);
		$this->day = $reader->readInteger(2);
		$this->hour = $reader->readInteger(3);
		$this->minute = $reader->readInteger(4);
		$this->second = $reader->readInteger(5);
	}

	public function write($writer)
	{
		$writer->writeValue(0, $this->year);
		$writer->writeValue(1, $this->month);
		$writer->writeValue(2, $this->day);
		$writer->writeValue(3, $this->hour);
		$writer->writeValue(4, $this->minute);
		$writer->writeValue(5, $this->second);
	}
}
?>

File: Sitana.Framework/CrashReporter/Data/DateAndTime.cs
using System;
using Sitana.Framework.DataTransfer;

namespace Sitana.Framework.CrashReporter.Data
{
	public class DateAndTime: IEcsStructure
	{
		public int Year;
		public int Month;
		public int Day;
		public int Hour;
		public int Minute;
		public int Second;

        public DateAndTime()
        {
        }

        public DateAndTime(DateTime dateTime)
        {
            Year = dateTime.Year;
            Month = dateTime.Month;
            Day = dateTime.Day;

            Hour = dateTime.Hour;
            Minute = dateTime.Minute;
            Second = dateTime.Second;
        }

        void IEcsStructure.Read(EcsReader reader)
		{
			Year = reader.ReadInt32(0);
			Month = reader.ReadInt32(1);
			Day = reader.ReadInt32(2);
			Hour = reader.ReadInt32(3);
			Minute = reader.ReadInt32(4);
			Second = reader.ReadInt32(5);
		}

        void IEcsStructure.Write(EcsWriter writer)
		{
			writer.Write(0, Year);
			writer.Write(1, Month);
			writer.Write(2, Day);
			writer.Write(3, Hour);
			writer.Write(4, Minute);
			writer.Write(5, Second);
		}
	}
}